meshpy/element_beam.py

```diff
@@ -125,7 +122,8 @@ def check_node(node, pos, rot, name):
             if (i > 0 or not has_start_node) and (
                 i < len(self.nodes_create) - 1 or not has_end_node
             ):
-                self.nodes.append(NodeCosserat(pos, rot, is_middle_node=middle_node))
+                is_middle_node = not (i == 0 or i == len(self.nodes_create) - 1)
```

Comment:
And just an idea - maybe that's easier to read - you can also merge
```suggestion
                is_middle_node = 0 < i < len(self.nodes_create) - 1
```